client/src/lib.rs

```diff
@@ -0,0 +1,154 @@
+//! A high-level client for interacting with a Celestia node.
+//!
+//! It combines the functionality of [`celestia-rpc`] and [`celestia-grpc`] crates.
+//!
+//! There are two modes: read-only mode and submit mode. Read-only mode requires
+//! RPC endpoint and submit mode requires RPC/GRPC endpoints, and a signer.
+//!
+//! # Examples
+//!
+//! Read-only mode:
+//!
+//! ```no_run
+//! # use celestia_client::{Client, Result};
+//! # const RPC_URL: &str = "http://localhost:26658";
+//! # async fn docs() -> Result<()> {
+//! let client = Client::builder()
+//!     .rpc_url(RPC_URL)
+//!     .build()
+//!     .await?;
+//!
+//! client.header().head().await?;
+//! # Ok(())
+//! # }
+//! ```
+//!
+//! Submit mode:
+//!
+//! ```no_run
+//! # use celestia_client::{Client, Result};
+//! # use celestia_client::tx::TxConfig;
+//! # const RPC_URL: &str = "http://localhost:26658";
+//! # const GRPC_URL : &str = "http://localhost:19090";
+//! # async fn docs() -> Result<()> {
+//! let client = Client::builder()
+//!     .rpc_url(RPC_URL)
+//!     .grpc_url(GRPC_URL)
+//!     .plaintext_private_key("...")?
+//!     .build()
+//!     .await?;
+//!
+//! let to_address = "celestia169s50psyj2f4la9a2235329xz7rk6c53zhw9mm".parse().unwrap();
+//! client.state().transfer(&to_address, 12345, TxConfig::default()).await?;
+//! # Ok(())
+//! # }
+//! ```
+//!
+//! [`celestia-rpc`]: celestia_rpc
+//! [`celestia-grpc`]: celestia_grpc
+
+mod blob;
+mod blobstream;
+mod client;
+mod fraud;
+mod header;
+mod share;
+mod state;
+mod utils;
+
+/// API related types.
+pub mod api {
+    pub use crate::blob::BlobApi;
+    pub use crate::blobstream::BlobstreamApi;
+    pub use crate::fraud::FraudApi;
+    pub use crate::header::HeaderApi;
+    pub use crate::share::ShareApi;
+    pub use crate::state::StateApi;
+
+    /// Blob API related types.
+    pub mod blob {
+        #[doc(inline)]
+        pub use crate::blob::BlobsAtHeight;
+    }
+
+    /// Share API related types.
+    pub mod share {
+        #[doc(inline)]
+        pub use crate::share::{GetRangeResponse, GetRowResponse, RowSide, SampleCoordinates};
+    }
```

Comment:
These are some types that exists in `celestia-rpc`, `celestia-grpc`, `celestia-client` but not in `celestia-types`.

Should I re-export types from `celestia-types` too?

Reply:
`SampleCoordinates` is an input type. I prefer to expose the output type in case users want to wrap our client